### DISCUSSION TITLE
Removing ROS_WORLD_SKIP_GROUPS for specific package

### DISCUSSION THREAD

**Discussion by bchoineubility** [bmwcarit/meta-ros#1112] [Q&A]
Hi,

I have a question for removing **ROS_WORLD_SKIP_GROUPS** for specific package  described in ros-distro.inc. 

For example, if I can provide openni recipe,  can I build image without any additional modification for meta-ros?
Or should I start from [Superflore-OE-Recipe-Generation-Scheme](https://github.com/ros/meta-ros/wiki/Superflore-OE-Recipe-Generation-Scheme)  ?

BR,
Mark

**Comment by robwoolley** [comment]
Hi Mark,

The ROS_WORLD_SKIP_GROUP is simply a way to add keywords that remove groups of packages from the dependencies of the packagegroup-ros-world-<DISTRO> recipe.   It will only be active if the keyword is set in ros-distro.inc if the recipes have outstanding issues eg. https://github.com/ros/meta-ros/blob/81d4e9de7a2219a55fb8d6484bff4e176f5734dd/meta-ros2-foxy/conf/ros-distro/include/foxy/ros-distro.inc#L54 or if you decide to set it in conf/local.conf.  It is needed for the world builds, but I think the better approach would be to create new packagegroups and images (#1068) so you can build up with a list of packages you do want rather than removing the ones you don't.

With respect to your example of openni. There are recipes for [openni](https://github.com/ros/meta-ros/blob/master/meta-ros2/recipes-support/openni/openni_1.5.4.0.bb) and [openni2](https://github.com/ros/meta-ros/blob/master/meta-ros2/recipes-support/openni2/openni2_2.2.0.bb).   I removed setting ROS_WORLD_SKIP_GROUP for openni or openni2 in ros-distro.inc for the supported releases.   However, openni2 and openni2-camera would get removed if "opengl" is added to ROS_WORLD_SKIP_GROUP: eg. https://github.com/ros/meta-ros/blob/master/meta-ros2-humble/recipes-core/packagegroups/packagegroup-ros-world-humble.bb#L312

It is a little confusing, but I hope that helps.

Regards,
Rob

**Reply by bchoineubility** [reply]
Thanks for clarification.
